From 569f07202ccf34fe630da60c794c8c60ca720ffe Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Thu, 23 Dec 2021 14:03:46 -0600
Subject: [PATCH 2/2] Revert "of: fix missing kobject init for !SYSFS &&
 OF_DYNAMIC config"

This reverts commit c6c4c74a0fe1bbe0141311f1bcec9e175897fd99.

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 drivers/of/base.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/of/base.c b/drivers/of/base.c
index c0281be8e061..41b254be0295 100644
--- a/drivers/of/base.c
+++ b/drivers/of/base.c
@@ -169,6 +169,9 @@ int __of_attach_node_sysfs(struct device_node *np)
 	struct property *pp;
 	int rc;
 
+	if (!IS_ENABLED(CONFIG_SYSFS))
+		return 0;
+
 	if (!of_kset)
 		return 0;
 
-- 
2.30.2

